pkg/kubectl/util/hash/hash.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hash
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"sort"
+
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// ConfigMapHash returns a hash of the EncodeConfigMap encoding of `cm`
+func ConfigMapHash(cm *api.ConfigMap) string {
+	return encodeHash(hash(encodeConfigMap(cm)))
+}
+
+// SecretHash returns a hash of the EncodeSecret encoding of `sec`
+func SecretHash(sec *api.Secret) string {
+	return encodeHash(hash(encodeSecret(sec)))
+}
+
+// encodeConfigMap encodes a ConfigMap by encoding the Data with EncodeMapStringString and wrapping with {} braces
+func encodeConfigMap(cm *api.ConfigMap) string {
+	return fmt.Sprintf("{data:%s,}", encodeMapStringString(cm.Data))
```

Comment:
including the pre-hash name and the kind (ConfigMap) would prevent comparison between two hash suffixes based solely on name (e.g. you could deduce configmaps `foo-014fbff9a07c` and `bar-014fbff9a07c` likely contain identical data based only on their names)